src/apps/search-result/search-result.tsx

```diff
@@ -49,11 +48,19 @@ const SearchResult: React.FC<SearchResultProps> = ({ q, pageSize }) => {
   };
 
   // If q changes (eg. in Storybook context)
-  //  then make sure that we reset the entire result set.
+  // then make sure that we reset the entire result set.
   useEffect(() => {
     setResultItems([]);
   }, [q, pageSize, filters]);
 
+  // Tracking whenever the search query changes.
+  const { track } = useStatistics();
+  useEffect(() => {
+    track({ internalSearch: q });
+    // We actaully just want to track if the query changes.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [q]);
```

Comment:
Based on our discussion today this may not be the actual implementation of search tracking that we want to use.

I do not know whether we should let it stay in the current form or await more clarification.

Reply:
I adjusted it, but I think we should keep it for testing 😉